### ISSUE TITLE
dtype mismatches in reconstructing patches

### ISSUE THREAD

**Issue by wsuzume** [hacarus/spm-image#36]
There is a bug in `feature_extraction.reconstruct_from_simple_patches_2d`. It can't reconstruct when type of `shape` argument is `Tuple[int, int, int]`, even though `extract_simple_patches_2d` seems to be working correctly in case of being given the same type.

**Comment by tksmd** [comment]
@wsuzume The following code seems to test such case. Could you share detail about the failure you encountered ?

* https://github.com/hacarus/spm-image/blob/development/tests/test_feature_extraction_image.py#L41-L49

**Comment by wsuzume** [comment]
@tksmd Thank you for giving me an information for tests.

I carefully checked and found simple mistakes in my code, that was using `sklearn.feature_extraction.image .extract_patches_2d` for patch extraction but was using `spmimage.feature_extraction.image.reconstruct_from_simple_patches_2d` for reconstruction. I'm sorry for that. After I fixed my code, the test is going successfully and matrices are reconstructed correctly.

But I noticed that matrix's `dtype` is chaged comparing before `extract_simple_patches_2d` and after `reconstruct_from_simple_patches_2d`. I will add some assertion to unittests, and will push some fixes to `issue/36/reconstruct-fails`.
